crownstone_uart/core/CrownstoneUart.py

```diff
@@ -66,21 +69,31 @@ def handleMessage(result, data):
         self.uartManager.start()
 
         while not result[0] and self.running:
+            try:
+                exc = self.manager_exception_queue.get(block=False)
+            except queue.Empty:
+                pass
+            else:
+                _LOGGER.warning(f"Error occurred while initializing USB: {exc}, quitting.")
+                # wait for thread to close
+                self.uartManager.join()
+                self.stop()
+                break
```

Comment:
Queue was used to handle multiple exceptions, there is only one now, but in case we want to handle more in the future, you can get the exception info from all those exceptions and not just one.

You're right about the else block, I will change it